builder/vsphere/common/step_shutdown.go

```diff
@@ -56,9 +57,22 @@ func (s *StepShutdown) Run(ctx context.Context, state multistep.StateBag) multis
 		return multistep.ActionContinue
 	}
 
-	if s.Config.DisableShutdown {
+	if state.Get("communicator") == nil {
+
+		ui.Say("The `communicator` is 'none', automatic shutdown disabled.")
+		if s.Config.Command != "" {
+			ui.Message("The parameter `shutdown_command` is ignored as it requires a `communicator`.")
+		}
```

Comment:
I think this can go inside `ShutdownConfig.Prepare()`
You will have to change the method signature and return to receive the `Comm` config and return a warning. 
Example of how it will look like: 
```
func (c *ShutdownConfig) Prepare(comm communicator.Config) (warnings []string, errs []error) {
	if c.Timeout == 0 {
		c.Timeout = 5 * time.Minute
	}
	if comm.Type == "none" && c.Command != "" {
		warnings = append(warnings, "The parameter `shutdown_command` is ignored as it requires a `communicator`.")
	}
	return
}
```
And in builder/vsphere/iso/config.go you will update the  `ShutdownConfig.Prepare()` call to:
```
	shutdownWarnings, shutdownErrs := c.ShutdownConfig.Prepare(c.Comm)
	warnings = append(warnings, shutdownWarnings...)
	errs = packer.MultiErrorAppend(errs, shutdownErrs...)
```

Reply:
I learned enough Golang to work on this issue so this is still pretty far out of my understanding.

For the `Prepare()` method it appears that this is receiving the communicator object as the parameter and then adding the the ability to handle warnings to the existing handling of errors.

The `return` line though was originally `return errs`. What is returning at this point? Is that defined by this?
```go
func (c *ShutdownConfig) Prepare() (warnings []string, errs []error) {
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Then assigned here? (to then be appended to the global vars)
```go
shutdownWarnings, shutdownErrs := c.ShutdownConfig.Prepare(c.Comm)
```

Mmmm... I think I understand the rest. At least conceptually. :smile:

Reply:
Ellipses tells a variable to iterate. Interesting.

Would the import statement still need the communicator? I believe the answer is yes.

Reply:
```bash
user@vm1:~/go/src/github.com/hashicorp/packer$ go build -o bin/packer .# github.com/hashicorp/packer/builder/vsphere/clone
builder/vsphere/clone/config.go:73:63: not enough arguments in call to c.ShutdownConfig.Prepare
        have ()
        want (communicator.Config)
builder/vsphere/clone/config.go:73:63: multiple-value c.ShutdownConfig.Prepare() in single-value context
user@vm1:~/go/src/github.com/hashicorp/packer$
```

Need to update `builder/vsphere/clone/config.go` for the modified `Prepare()`...

Reply:
I like that more. It puts the warning at the top.